src/error.rs

```diff
@@ -88,4 +90,11 @@ impl<T: Sized> From<PoisonError<T>> for Error {
     }
 }
 
+impl From<Box<dyn Any + Send>> for Error {
```

Comment:
I don'e think we should be able to convert arbitrary `Box<dyn Any + Send>` to Error. We should explicitly write `.map_err(Error::JoinError)` when necessary.